use regexp for more lenient opened enhancement matching

nmschulte opened the pull request:
for #8

I've never Go'ne before; please review the regexp addition.  May be applicable to other auto labeling-from-keyword scenarios, but I didn't see any.

stapelberg commented on bot.go:
```diff
@@ -407,11 +407,9 @@ func issuesHandler(w http.ResponseWriter, r *http.Request) {
 	lcBody := strings.ToLower(*payload.Issue.Body)
 	lcTitle := strings.ToLower(*payload.Issue.Title)
 
+	enhancementRegexp = regexp.MustCompile("feature.?request|enhancement")
```
It is customary to initialize regexps outside of functions so that they’re only compiled once.

stapelberg reviewed (commented):
I’m not quite sure why you’re proposing the code change. AFAICT, it’s a no-op which only makes the code slower (regexps are slower than 2 strings.Contains calls)?

nmschulte commented:
The change to a regular expression allows a optional (any) character between `feature` and `request`.

e.g. https://github.com/i3/i3/issues/2733 would have been caught (title'd `[feature-request]`)

I won't mind either way; this is me covering for my issue request (#8, where "this feature already exists").

stapelberg commented:
Ah, thanks for explaining. In that case, instead of describing the how (“use regexps”), please make your commit message describe the why (“match "feature request" more leniently”).

nmschulte committed:
match new enhancement more leniently

use a regular expression to match enhancement-like strings;
/feature.?request/ -- previous code used string matching with multiple tokens in a string ("feature request")

nmschulte commented:
updated commit message -- I'm not sure my `const enhancementRegexp ...` (now at top-level scope) syntax is proper

stapelberg commented:
Thanks!

stapelberg commented:
It’ll be a while before I can deploy this: our dependency `github.com/google/go-github` started using `context` which was only introduced in Go 1.7, but AppEngine still runs Go 1.6.
